packages/smooth_app/integration_test/app_test.dart

```diff
@@ -0,0 +1,89 @@
+import 'dart:io';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:integration_test/integration_test.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+import 'package:smooth_app/main.dart' as app;
+
+Future<void> _initScreenshot(
+  final IntegrationTestWidgetsFlutterBinding binding,
+) async {
+  if ((!kIsWeb) && Platform.isAndroid) {
+    await binding.convertFlutterSurfaceToImage();
+  }
+}
+
+Future<void> _takeScreenshot(
+  final WidgetTester tester,
+  final IntegrationTestWidgetsFlutterBinding binding,
+  final String screenshotName,
+) async {
+  if ((!kIsWeb) && Platform.isAndroid) {
+    await tester.pumpAndSettle();
+  }
+  await binding.takeScreenshot(screenshotName);
+}
+
+// flutter drive --driver=test_driver/screenshot_driver.dart --target=integration_test/app_test.dart
+
+/// Onboarding screenshots.
+void main() {
+  final IntegrationTestWidgetsFlutterBinding binding =
+      IntegrationTestWidgetsFlutterBinding.ensureInitialized()
+          as IntegrationTestWidgetsFlutterBinding;
+
+  setUpAll(
+    () => SharedPreferences.setMockInitialValues(
+      const <String, Object>{
+        'IMPORTANCE_AS_STRINGnutriscore': 'important',
+        'IMPORTANCE_AS_STRINGnova': 'important',
+        'IMPORTANCE_AS_STRINGecoscore': 'important',
+      },
+    ),
+  );
+
+  group('end-to-end test', () {
+    testWidgets('just a single screenshot', (WidgetTester tester) async {
+      await tester.runAsync(() async {
+        await _initScreenshot(binding);
+
+        await app.main(screenshots: true);
+        await tester.pumpAndSettle();
+
+        sleep(const Duration(seconds: 30));
```

Comment:
I think `pumpAndSettle` already takes care of waiting

Reply:
I'm not sure of anything anymore. Especially when there's async load: ok the `pumpAndSettle` will tell you that the display is refreshed, but not that the data was loaded async'ly in order to refresh the display.